runtime-modules/storage/src/lib.rs

```diff
@@ -737,6 +745,18 @@ pub struct Voucher {
     pub objects_used: u64,
 }
 
+impl Voucher {
+    fn update(self, new_voucher: VoucherUpdate) -> Result<Self, &'static str> {
+        ensure!(new_voucher.objects_number <= self.objects_limit, "number");
```

Comment:
Please, remove "magic consts" and consider placing `ensure` checks in the calling code or as a separate function.

Reply:
:+1: